mbuild/formats/gsdwriter.py

```diff
@@ -68,44 +68,46 @@ def write_gsd(structure, filename, ref_distance=1.0, ref_mass=1.0,
         yz = (b*c*np.cos(alpha) - xy*xz) / ly
         lz = np.sqrt(c**2 - xz**2 - yz**2)
 
-        gsd_file.configuration.box = np.array([lx, ly, lz, xy, xz, yz])
+        gsd_snapshot.configuration.box = np.array([lx, ly, lz, xy, xz, yz])
 
-    _write_particle_information(gsd_file, structure, xyz, ref_distance,
+    _write_particle_information(gsd_snapshot, structure, xyz, ref_distance,
             ref_mass, ref_energy, rigid_bodies)
     if write_special_pairs:
-        _write_pair_information(gsd_file, structure)
+        _write_pair_information(gsd_snapshot, structure)
     if structure.bonds:
-        _write_bond_information(gsd_file, structure)
+        _write_bond_information(gsd_snapshot, structure)
     if structure.angles:
-        _write_angle_information(gsd_file, structure)
+        _write_angle_information(gsd_snapshot, structure)
     if structure.rb_torsions:
-        _write_dihedral_information(gsd_file, structure)
+        _write_dihedral_information(gsd_snapshot, structure)
 
 
-    gsd.hoomd.create(filename, gsd_file)
+    gsd_file = gsd.hoomd.open(filename, mode='wb')
```

Comment:
Since this is a file pointer like object, we should use a context manager to handle automatic closing of the filestream.

See https://github.com/mosdef-hub/mbuild/pull/683/files#diff-53fbfac5b93a85f9c95e9c7fca5891f2R84